mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,195 @@
+[P0] AUC: Approximates the AUC (Area under the curve) of the ROC or PR curves.
+[P0] F1 Score: 2*precision*recall / (precision+recall)
+
+[P0] BinaryClassConfusionMatrix
+
+Plots
+[P0] Confusion matrix
+[P0] Interactive ROC curve with metrics (TP/TN/FP/FN/Acc/F1/AUC), binary classification
+[P0] Lift chart
+
+Global explainability
+[P0] Model built-in feature importance (supported models)
+[P0] SHAP explainers
+     [P0] Summary plot
+"""
+
+from PIL.Image import Image, open as open_image
+
+
+class ImageEvaluationArtifact(EvaluationArtifact):
+
+    @classmethod
+    def save_content_to_file(cls, content, output_artifact_path):
+        assert isinstance(content, Image)
+        content.save(output_artifact_path)
+
+    @classmethod
+    def load_content_from_file(cls, local_artifact_path):
+        return open_image(local_artifact_path)
+
+
+class DefaultEvaluator(ModelEvaluator):
+
+    def can_evaluate(self, model_type, evaluator_config=None, **kwargs):
+        return model_type in ["classifier", "regressor"]
+
+    def _log_image_artifact(self, artifacts, temp_dir, do_plot, run_id, artifact_name, dataset_name):
+        pyplot.clf()
+        do_plot()
+        artifact_file_name = f"{artifact_name}_on_{dataset_name}.png"
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pyplot.savefig(artifact_file_local_path)
+        mlflow.log_artifact(artifact_file_local_path, artifact_file_name)
+        artifacts[artifact_file_name] = \
+            ImageEvaluationArtifact(
+                location=get_artifact_uri(run_id, artifact_file_name),
+                content=ImageEvaluationArtifact.load_content_from_file(artifact_file_local_path)
+            )
+
+    def _log_model_explainality(
+        self, artifacts, temp_dir, model, X, dataset_name, run_id
+    ):
+        explainer = shap.Explainer(model, X)
+        shap_values = explainer(X)
+
+        def plot_summary():
+            shap.plots.beeswarm(shap_values, show=False)
+
+        self._log_image_artifact(artifacts, temp_dir, plot_summary, run_id, 'shap_summary', dataset_name)
+
+        def plot_feature_importance():
+            shap.plots.bar(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts, temp_dir, plot_feature_importance, run_id, 'shap_feature_importance', dataset_name
+        )
+
+    def _classifier_compute_metrics_and_compute_and_log_artifacts(
+        self, temp_dir, model, X, y, dataset_name, run_id, evaluator_config
+    ):
+        # TODO: how to get num_classes without extra config ?
+        num_classes = evaluator_config['num_classes']
+        y_pred = model.predict(X)
+
+        is_binomial = (num_classes <= 2)
+
+        metrics = EvaluationMetrics()
+        artifacts = {}
+        metrics['accuracy'] = sk_metrics.accuracy_score(y, y_pred)
+        metrics['example_count'] = len(X)
+
+        # TODO: sum/mean on what data ?
+        #  [P0] Sum: Computes the (weighted) sum of the given values.
```

Comment:
I've only calculated these as precursors to other measurement metrics and only for regression problems. Classification doesn't really have anything relevant with summing of encoded labels or getting the mean of their arbitrary values.

Reply:
for regressor, we log mse, mae, rmse, I think we don't need log sum/mean for regressor

Reply:
100% agree.